frontend/src/Components/modals/VerfiyResidentModal.tsx

```diff
@@ -0,0 +1,135 @@
+import { forwardRef, useState } from 'react';
+import { CRUDModalProps, FormInputTypes, FormModal, Input } from '.';
+import { FieldValues, SubmitHandler } from 'react-hook-form';
+import {
+    Facility,
+    ResidentEngagementProfile,
+    ServerResponseOne,
+    ValidResident
+} from '@/common';
+import API from '@/api/api';
+import { useCheckResponse } from '@/Hooks/useCheckResponse';
+import useSWR from 'swr';
+
+export const VerifyResidentModal = forwardRef(function (
+    {
+        mutate,
+        onSuccess,
+        target
+    }: CRUDModalProps<ResidentEngagementProfile> & {
+        onSuccess: (resident: ValidResident) => void;
+    },
+    verifyResidentModal: React.ForwardedRef<HTMLDialogElement>
+) {
+    const [errors, setErrors] = useState('');
+    const [facility, setFacility] = useState('');
+    const { data: facilitiesData } =
+        useSWR<ServerResponseOne<Facility[]>>('/api/facilities');
+    const facilities = facilitiesData?.data;
+    const checkResponse = useCheckResponse({
+        mutate: mutate,
+        refModal: verifyResidentModal
+    });
+    const executeResidentCheck: SubmitHandler<FieldValues> = async (data) => {
+        if (!validateFacility()) {
+            return;
+        }
+        const response = (await API.get(
+            `users/resident-verify?user_id=${target?.user.id}&doc_id=${data.doc_id}&facility_id=${facility}`
+        )) as ServerResponseOne<ValidResident>;
+        checkResponse(
+            response.success,
+            'Resident not found, try again',
+            'Resident found, please confirm transfer'
```

Comment:
Can we get by with out a success toast message since the presence of the TransferSummary panel serves the same purpose?